src/js/uart.js

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2016 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var EventEmitter = require('events').EventEmitter;
+var util = require('util');
+var assert  = require('assert');
+
+var uart = process.binding(process.binding.uart);
+
+//  VALIDATION ARRAYS
+var DATABITS = [5, 6, 7, 8];
+var STOPBITS = [1, 1.5, 2];
+var PARITY = ['none', 'even', 'mark', 'odd', 'space'];
+var FLOWCONTROLS = ['xon', 'xoff', 'xany', 'rtscts'];
+
+var defaultSettings = {
+  autoOpen: true,
+  baudRate: 9600,
+  dataBits: 8,
+  hupcl: true,
+  lock: true,
+  parity: 'none',
+  rtscts: false,
+  stopBits: 1,
+  xany: false,
+  xoff: false,
+  xon: false,
+  highWaterMark: 16 * 1024,
+}
+
+function UART(path, options, callback) { //constructor
+  if (!(this instanceof UART)) {
+    return new UART(path, options, callback);
+  }
+
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+
+  if (util.isUndefined(options)) {
+    options = {};
+  }
+
+  var settings = defaultSettings;
+  for (var propname in options) {
+    settings[propname] = options[propname];
+  }
+
+  EventEmitter.call(this);
+
+  if (!path) {
+    throw new TypeError('No path specified');
+  }
+
+  if (typeof settings.baudRate !== 'number') {
+    throw new TypeError("'baudRate' must be a number");
+  }
+
+  if (DATABITS.indexOf(settings.dataBits) === -1) {
+    throw new TypeError("Invalid 'databits': " + settings.dataBits);
+  }
+
+  if (STOPBITS.indexOf(settings.stopBits) === -1) {
+    throw new TypeError("Invalid 'stopbits': " + settings.stopbits);
+  }
+
+  if (PARITY.indexOf(settings.parity) === -1) {
+    throw new TypeError("Invalid 'parity': " + settings.parity);
+  }
+
+  FLOWCONTROLS.forEach(function(control) {
+    if (typeof settings[control] !== 'boolean') {
+      throw new TypeError("Invalid '" + control + "' is not boolean");
+    }
+  });
+
+  this.path = path;
+  this.options = settings;
+  this.opening = false;
```

Comment:
I can't fine use ``this.opening`` variable in whole js code.(Only assign a value in variable) Is it necessary? Or is this variable is for the future implementation?